packages/database/src/DatabaseModel.js

```diff
@@ -18,17 +18,29 @@ export class DatabaseModel {
     this.schema = null;
 
     this.cache = {};
+    this.cachedFunctionInvalidationCancellers = {};
 
     // If this model uses the singleton database, it is probably long running, so be sure to
     // invalidate the cache any time a change is detected. Non-singleton models are those created
     // during transactions, so are short lived and unlikely to need cache invalidation - thus we
     // avoid making an additional connection to pubsub and just leave their cache untouched
     if (this.database.isSingleton) {
+      // fully reset cache on any change to this model's records
       this.database.addChangeHandlerForCollection(this.DatabaseTypeClass.databaseType, () => {
-        this.cache = {}; // invalidate cache on any change
+        this.cache = {};
       });
+
+      // if this model has caching that depends on other models, also add invalidation for them
+      if (this.cacheDependencies) {
+        this.cacheDependencies.forEach(databaseType => {
```

Comment:
A minor suggestion, I see you used a getter for this property in `Entity.js`. Could we then define a base value for this property in `DatabaseModel.js`, i.e.

```js
  get cacheDependencies() {
    return [];
  }
```

or `this.cacheDependencies = [];`

In that case, we can avoid checking `if (this.cacheDependencies)`

Reply:
Good idea :-)